### ISSUE TITLE
Feedback on `multi_download()`

### ISSUE THREAD

**Issue by hadley** [jeroen/curl#286]
I tried it out on my motivating use case, a large csv file I use in R4DS (https://r4ds.s3.us-west-2.amazonaws.com/seattle-library-checkouts.csv), and another example I made to try it with multiple files:

```R
host <- "https://d37ci6vzurychx.cloudfront.net/"
urls <- paste0(host, "trip-data/yellow_tripdata_2021-", sprintf("%02d", 1:12), ".parquet")

dir <- tempfile()
dir.create(dir)
files <- file.path(dir, basename(urls))

curl::multi_download(urls, files, resume = TRUE, progress = TRUE)
```

Overall, it worked great, and it seemed to happily resume when I interrupted it.

Some feedback/ideas (many of which are probably out of scope, but these are the things that occurred to me):

* I think rounding to the nearest MB would be fine; the extra decimal places give a lot of jitter
* It would be nice to somehow report how many requests were resumed.
* Maybe it would be useful to include the existing file size in the total downloaded? (i.e. total downloaded including resumes). Or maybe have a different summary for that?
* I think if the file is resumable, it's likely that you know the total size? So it would be great to report the percentage downloaded and an ETA. I know it's hard to manage progress bars for multiple files when we only have a single line, but it's really nice to know that it has successfully resumed.
* Could you report the current download speed? 
* Presumably if the file is resumable and you know the total size, you could request multiple chunks in parallel? 
* Would be useful if the docs described what the columns in the output mean. e.g. what are the units for `time`?
* It'd be useful to still return the data frame values even if interrupted or errored. i.e. wrap `multi_run()` in a `tryCatch()` that captures both errors and interrupts.
* Are you sure `progress` and `resume` shouldn't default to `TRUE`? Or maybe `progress` should default to `interactive()`? Or have a different progress bar for batch usage that prints once every 10s?

**Comment by jeroen** [comment]
I have improved the progress function to include speed and percent complete (if available).

**Comment by hadley** [comment]
Looks great!

I'd still love to see some messaging on what files we resumed, and I think you could also round the transfer speed (and maybe give it a fixed width), but don't see any other obvious deficiencies.

**Comment by jeroen** [comment]
Do you think the name `multi_download()` is OK?

**Comment by hadley** [comment]
Hmmmm, maybe not since most of the other `multi_` functions seem to be for advanced use only.  Maybe `curl_download_many()` or similar?
